js/apps/admin-ui/src/clients/authorization/policy/PolicyDetails.tsx

```diff
@@ -164,9 +167,17 @@ export default function PolicyDetails() {
     return <KeycloakSpinner />;
   }
 
-  const ComponentType = isValidComponentType(policyType)
-    ? COMPONENTS[policyType]
-    : COMPONENTS["js"];
+  function getComponentType() {
+    if (isDisabled) {
+      return COMPONENTS["js-disabled"];
```

Comment:
I was thinking, rather than handling `js-disable` as an edge case like this. Perhaps we should add a `isDisabled` prop to the `COMPONENTS`? That way we can re-use the existing `JavaScript` component.

Reply:
Hi @jonkoops , we are just checking the above edge case in the `getComponentType() `function to assign a ComponentType . For other components like `NameDescription`,`LogicSelector` and `Button` we are passing `isDisabled` as a prop.

Reply:
I believe that @ssilvert mentioned to me that the JS one should always be disabled. But I do not recall the fine details.

Reply:
@jonkoops @ssilvert If we disable the JS component, then the user will not have the possibility to create a new JS based policy as the input box will always be disabled. As of `js` points to the `JavaScript` component and `js-disabled` points to the `JavaScriptDisabled` component which it seems to be working fine.

Reply:
I have verified this now.  Awhile back, we decided that javascript should never be "uploadable".   Furthermore, you should not be able to create a javascript policy from the UI.

Reply:
In that case I think we can make the default `js` read only, that should be enough. It also removes the need to explicitly check the name of the default policy.

Reply:
@jonkoops @ssilvert ack , thanks for the confirmation